packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -367,276 +471,278 @@ private void createCaptureSessionWithSessionConfig(
-          break;
-        case always:
-        default:
-          captureBuilder.set(
-              CaptureRequest.CONTROL_AE_MODE, CaptureRequest.CONTROL_AE_MODE_ON_ALWAYS_FLASH);
-          break;
-      }
-      cameraCaptureSession.stopRepeating();
-      cameraCaptureSession.capture(
-          captureBuilder.build(),
+          lockedOrientation == null
+              ? getDeviceOrientationManager().getPhotoOrientation()
+              : getDeviceOrientationManager().getPhotoOrientation(lockedOrientation));
+
+      CameraCaptureSession.CaptureCallback captureCallback =
           new CameraCaptureSession.CaptureCallback() {
+
+            @Override
+            public void onCaptureStarted(
+                @NonNull CameraCaptureSession session,
+                @NonNull CaptureRequest request,
+                long timestamp,
+                long frameNumber) {
+              Log.i(TAG, "onCaptureStarted");
+            }
+
+            @Override
+            public void onCaptureProgressed(
+                @NonNull CameraCaptureSession session,
+                @NonNull CaptureRequest request,
+                @NonNull CaptureResult partialResult) {
+              Log.i(TAG, "onCaptureProgressed");
+            }
+
             @Override
             public void onCaptureCompleted(
                 @NonNull CameraCaptureSession session,
                 @NonNull CaptureRequest request,
                 @NonNull TotalCaptureResult result) {
+              Log.i(TAG, "onCaptureCompleted");
               unlockAutoFocus();
             }
-          },
-          null);
+          };
+
+      captureSession.stopRepeating();
+      captureSession.abortCaptures();
+      Log.i(TAG, "sending capture request");
+      captureSession.capture(stillBuilder.build(), captureCallback, backgroundHandler);
     } catch (CameraAccessException e) {
-      pictureCaptureRequest.error("cameraAccess", e.getMessage(), null);
+      dartMessenger.error(flutterResult, "cameraAccess", e.getMessage(), null);
     }
   }
 
-  private void lockAutoFocus(CaptureCallback callback) {
-    captureRequestBuilder.set(
+  @SuppressWarnings("deprecation")
+  private Display getDefaultDisplay() {
+    return activity.getWindowManager().getDefaultDisplay();
+  }
+
+  /** Starts a background thread and its {@link Handler}. TODO: call when activity resumed */
+  private void startBackgroundThread() {
+    backgroundHandlerThread = new HandlerThread("CameraBackground");
+    backgroundHandlerThread.start();
+    backgroundHandler = new Handler(backgroundHandlerThread.getLooper());
+  }
+
+  /** Stops the background thread and its {@link Handler}. TODO: call when activity paused */
+  private void stopBackgroundThread() {
+    try {
+      if (backgroundHandlerThread != null) {
+        backgroundHandlerThread.quitSafely();
+        backgroundHandlerThread.join();
+        backgroundHandlerThread = null;
```

Comment:
if join throws, `backgroundHandlerThread` is leaked.  Please keep `join()` in the try block only, and reset these outside the try/catch block.

Reply:
I've pushed a fix for this, but I'm not 100% certain I fully understand the issue, so a double check would be appreciated.